### PR TITLE
Cross shard contract call

### PR DESCRIPTION
**Pull request by ailisp**
A pytest to replicate `Trying simple cross contract call` section (and all it's prerequisite step) from https://github.com/nearprotocol/near-bindgen/tree/master/examples/cross-contract-high-level

Test Plans
-------------
It should give described result as in the cross-contract-high-level link above

### CODE REVIEW

**Comment by evgenykuzyakov** [comment]
I think this is because you are acting on behalf of the `test0` when trying to deploy code on account `cross_contract`. You should send a transaction from `cross_contract` to deploy code.

You can also deploy contract code at the same time when you create the account as an additional action.

**Comment by ailisp** [comment]
Above issue was resolved by Evgeny and Illia's help. Now it's able to `deploy_status_message`, `simple_call`:
https://github.com/nearprotocol/near-bindgen/tree/master/examples/cross-contract-high-level
 but when do `get_status` it timesout even giving 480s. Is this my mistake or some bug? pin @bowenwang1996 for help as Alex suggest :) /cc @nearmax

**Comment by bowenwang1996** [comment]
How did you do `get_status` call?

**Comment by ailisp** [comment]
@bowenwang1996 I set sender to `status_message`, with content `{"test_id": "test0"} and sign with master account `test0`'s sk. It looks equivalent to how cross-contract-high-level did

**Comment by ailisp** [comment]
@bowenwang1996 Have some idea on why get_status timeout?

**Comment by bowenwang1996** [comment]
Will take a look. Somehow I forgot about this PR

**Comment by bowenwang1996** [comment]
@ailisp I made it work, but still not quite happy with the way we determine which shard the contract is deployed to (as it requires modifying nearcore code). Do you think it's better if we generate random suffix in the test and run the test multiple times so that probabilistically some run will trigger the case where the status message contract is deployed on a different shard?

**Comment by ailisp** [comment]
> @ailisp I made it work, but still not quite happy with the way we determine which shard the contract is deployed to (as it requires modifying nearcore code). Do you think it's better if we generate random suffix in the test and run the test multiple times so that probabilistically some run will trigger the case where the status message contract is deployed on a different shard?

Great! That's also Alex suggested and i planned to do finally. Was doing this simply as a simpler check point.

File: pytest/lib/transaction.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def sign_function_call_tx(signer_key, methodName, args, gas, deposit, nonce, blo
     if contractName is None:
         contractName = signer_key.account_id
     action = create_function_call_action(methodName, args, gas, deposit)
-    return sign_and_serialize_transaction(contractName, nonce, [action], blockHash, contractName, signer_key.decoded_pk(), signer_key.decoded_sk())
+    return sign_and_serialize_transaction(contractName, nonce, [action], blockHash, signer_key.account_id, signer_key.decoded_pk(), signer_key.decoded_sk())
```
Comment (ailisp) [inline pytest/lib/transaction.py]: Oh so this is the major mistake i made :+1:

**Comment by evgenykuzyakov** [comment]
@ailisp Let's close this branch.

**Pull request closed without merging**